File: dotnet/Scenarios/Exercises/Logic/Repository/Models/ChapterEntity.cs
namespace EncyclopediaGalactica.Scenarios.Exercises.Logic.Repository.Models;

public class ChapterEntity
{
    public long Id { get; set; }
    public string Title { get; set; }
    public string Reference { get; set; }
    public int PageStart { get; set; }
    public int PageEnd { get; set; }
    public long BookId { get; set; }
    public BookEntity Book { get; set; }

    public List<SectionEntity> Sections { get; set; } = [];
}

File: Rust/eg/src/logic/exercises/repository/entities/chapter_entity.rs
use sqlx::FromRow;

#[derive(Debug, FromRow)]
pub struct ChapterEntity {
    pub id: i32,
    #[sqlx(default)]
    pub title: String,
    #[sqlx(default)]
    pub reference: String,
    #[sqlx(default)]
    pub page_start: i32,
    #[sqlx(default)]
    pub page_end: i32,
    pub book_id: i32,
}
